zeppelin-web/src/app/credential/credential.controller.js

```diff
@@ -15,42 +15,135 @@
 'use strict';
 
 angular.module('zeppelinWebApp').controller('CredentialCtrl', function($scope, $route, $routeParams, $location, $rootScope,
-                                                                         $http, baseUrlSrv) {
+                                                                         $http, baseUrlSrv, ngToast) {
   $scope._ = _;
 
-  $scope.credentialEntity = '';
-  $scope.credentialUsername = '';
-  $scope.credentialPassword = '';
+  $scope.credentialInfo = [];
+  $scope.showAddNewCredentialInfo = false;
 
-  $scope.updateCredentials = function() {
-    if (_.isEmpty($scope.credentialEntity.trim()) ||
-        _.isEmpty($scope.credentialUsername.trim())) {
+  var getCredentialInfo = function() {
+    $http.get(baseUrlSrv.getRestApiBase()+'/credential').
+      success(function(data, status, headers, config) {
+
+        $scope.credentialInfo  = _.map(data.body.userCredentials, function(value, prop) {
+          return {entity: prop, password: value.password, username: value.username };
+        });
+        console.log('Success %o %o', status, $scope.credentialInfo);
+      }).
+      error(function(data, status, headers, config) {
+        console.log('Error %o %o', status, data.message);
+      });
+  };
+
+  $scope.addNewCredentialInfo = function() {
+    if ($scope.entity && _.isEmpty($scope.entity.trim()) &&
+      $scope.username && _.isEmpty($scope.username.trim())) {
       BootstrapDialog.alert({
         closable: true,
         message: 'Username \\ Entity can not be empty.'
       });
       return;
     }
 
-    $http.put(baseUrlSrv.getRestApiBase() + '/credential',
-      { 'entity': $scope.credentialEntity,
-        'username': $scope.credentialUsername,
-        'password': $scope.credentialPassword
-      } ).
+    var data  = {
+      'entity': $scope.entity,
+      'username': $scope.username,
+      'password': $scope.password
+    };
+
+    $http.put(baseUrlSrv.getRestApiBase() + '/credential', data).
     success(function (data, status, headers, config) {
       BootstrapDialog.alert({
         closable: true,
         message: 'Successfully saved credentials.'
       });
-      $scope.credentialEntity = '';
-      $scope.credentialUsername = '';
-      $scope.credentialPassword = '';
+      //getCredentialInfo();
```

Comment:
If we remove this line, it doesn't shown up in the table right after new credential info added. What do you think?